src/Provider/Doctrine/Auditing/Event/DoctrineSubscriber.php

```diff
@@ -4,21 +4,23 @@
 
 namespace DH\Auditor\Provider\Doctrine\Auditing\Event;
 
+use Closure;
 use DH\Auditor\Provider\Doctrine\Auditing\Logger\Middleware\DHDriver;
-use DH\Auditor\Provider\Doctrine\Auditing\Transaction\TransactionManager;
 use DH\Auditor\Provider\Doctrine\Model\Transaction;
+use DH\Auditor\Transaction\TransactionManagerInterface;
 use Doctrine\Common\EventSubscriber;
+use Doctrine\DBAL\Driver;
+use Doctrine\DBAL\Driver\Middleware\AbstractDriverMiddleware;
 use Doctrine\ORM\Event\OnFlushEventArgs;
 use Doctrine\ORM\Events;
+use ReflectionClass;
 
 final class DoctrineSubscriber implements EventSubscriber
 {
-    private TransactionManager $transactionManager;
+    /** @var Transaction[] */
+    private array $transactions = [];
 
-    public function __construct(TransactionManager $transactionManager)
-    {
-        $this->transactionManager = $transactionManager;
-    }
+    public function __construct(private readonly TransactionManagerInterface $transactionManager) {}
```

Comment:
this changes are require only for tests, because TransactionManager is final it cannot be mocked